File: Week 1/Mario/mario_more.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    // Variable definition
    int height, row, space, column, reversePy;

    //Gets input from user
    do
    {
        height = get_int("Pyramid height: ");
    }
    while (height < 1 || height > 8);

    // for each row, i= row
    for (row = 0; row < height; row++)
    {
        //For each space; space == k; space = height- row - 1 (cause we start counting at 0).
        for (space = 0; space < height - row - 1; space++)
        {
            printf(" ");
        }
        //for each column;
        for (column = 0; column <= row; column++)
        {
            printf("#");
        }
        // Loop to define the spaces between walls, using column = height.
        for (column = height; column == height || column == height + 1; column++)
        {
            printf(" ");
        }
        //Loop with reverse pyramid (reversePy <= row cause it will restart reversePy value based on Rows)
        for (reversePy = 0; reversePy <= row; reversePy++)
        {
            printf("#");
        }
        printf("\n");
    }

}

File: Week 6/Mario_Python/mario_more.py
# Get input from user
while True:
    try:
        height = int(input("Height: "))
        if 0 < height < 9:
            break
    except ValueError:
        print("Not an int.")
# Write pyramid using rows and columns
for row in range(height):
    for column in range(height):
        # If row+column == height-1 or row+column > height-1, we print a #.
        if row + column < height-1:
            print(" ", end="")
        else:
            print("#", end="")
    for space in range(height*2+1):
        # This pyramid grid size is defined by the formula 2*height+1, so this is the length of our loop.
        # Spaces are only needed when they get to the pyramid height or surpass it by 1.
        if space == height or space == height + 1:
            print(" ", end="")
    for column2 in range(height):
        if column2 <= row:
            print("#", end="")
    print()